client/test/commands/createSmartContractProjectCommand.test.ts

```diff
@@ -0,0 +1,265 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+'use strict';
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import * as tmp from 'tmp';
+import * as sinonChai from 'sinon-chai';
+chai.use(sinonChai);
+import * as fs from 'fs';
+import * as fs_extra from 'fs-extra';
+import * as util from 'util';
+import * as child_process from 'child_process';
+import { CommandUtil } from '../../src/util/CommandUtil';
+import { ExtensionUtil } from '../../src/util/ExtensionUtil';
+const mkdir = util.promisify(fs.mkdir);
+
+// Defines a Mocha test suite to group tests of similar kind together
+// tslint:disable no-unused-expression
+describe('CreateSmartContractProjectCommand', () => {
+    // suite variables
+    let mySandBox: sinon.SinonSandbox;
+    let sendCommandStub: sinon.SinonStub;
+    let errorSpy: sinon.SinonSpy;
+    let infoBoxSpy: sinon.SinonSpy;
+    let quickPickStub: sinon.SinonStub;
+    let openDialogStub: sinon.SinonStub;
+    let executeCommandStub: sinon.SinonStub;
+    let uri: vscode.Uri;
+    let uriArr: Array<vscode.Uri>;
+
+    beforeEach(async () => {
+        await ExtensionUtil.activateExtension();
+        mySandBox = sinon.createSandbox();
+        sendCommandStub = mySandBox.stub(CommandUtil, 'sendCommand');
+        errorSpy = mySandBox.spy(vscode.window, 'showErrorMessage');
+        infoBoxSpy = mySandBox.spy(vscode.window, 'showInformationMessage');
+        quickPickStub = mySandBox.stub(vscode.window, 'showQuickPick');
+        openDialogStub = mySandBox.stub(vscode.window, 'showOpenDialog');
+        const originalExecuteCommand = vscode.commands.executeCommand;
+        executeCommandStub = mySandBox.stub(vscode.commands, 'executeCommand');
+        executeCommandStub.callsFake(async function fakeExecuteCommand(command: string) {
+            // Don't open the folder as this causes lots of windows to pop up, and random
+            // test failures.
+            if (command !== 'vscode.openFolder') {
+                return originalExecuteCommand.apply(this, arguments);
+            }
+        });
+        uri = vscode.Uri.file(tmp.dirSync().name);
+        uriArr = [uri];
+    });
+    afterEach(() => {
+        mySandBox.restore();
+    });
+
+    // Define assertion
+    it('should start a typescript smart contract project', async () => {
+        // We actually want to execute the command!
+        sendCommandStub.restore();
+        try {
+            await mkdir(uri.fsPath);
+        } catch (error) {
+            if (!error.message.includes('file already exists') ) {
```

Comment:
I'm suspicious. If the temporary directory already exists, the contents could interfere with test results, e.g. tests for files that should exist